src/common/meta/src/key/flow/flow_name.rs

```diff
@@ -237,6 +237,39 @@ impl FlowNameManager {
             TxnOpGetResponseSet::decode_with(TxnOpGetResponseSet::filter(raw_key)),
         ))
     }
+
+    /// Builds a update flow name transaction.
+    /// It's expected that the `__flow/name/{catalog}/{flow_name}` IS already occupied.
+    /// Otherwise, the transaction will retrieve existing value(and fail).
+    pub fn build_update_txn(
+        &self,
+        catalog_name: &str,
+        flow_name: &str,
+        flow_id: FlowId,
+    ) -> Result<(
+        Txn,
+        impl FnOnce(&mut TxnOpGetResponseSet) -> FlowNameDecodeResult,
+    )> {
+        let key = FlowNameKey::new(catalog_name, flow_name);
+        let raw_key = key.to_bytes();
+        let flow_flow_name_value = FlowNameValue::new(flow_id);
+        let txn = Txn::new()
+            .when(vec![Compare::new(
+                raw_key.clone(),
+                CompareOp::NotEqual,
+                None,
+            )])
```

Comment:
Compare with the previous value?

Reply:
> Compare with the previous value?

I want to make sure the value exists before update txn, if set to compare with previous value then wouldn't it allow the value to be non-existing?